.github/workflows/build-test-and-publish.yml

```diff
@@ -0,0 +1,188 @@
+name: Build
+
+on:
+  push:
+    branches:
+      - master
+      - DEV-100_deployment-on-aws
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    name: Build Elixir project
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set version number
+        id: version
+        uses: DEalog/version-action@v0.1.0
+        with:
+          fallback-prefix: "1.0.0-SNAPSHOT-"
+
+      - name: Use cached node_modules
+        id: cache
+        uses: actions/cache@v1
+        with:
+          path: ./assets/node_modules
+          key: nodeModules-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            nodeModules-
+      
+      - name: Use Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: '14'
+
+      - name: Install dependencies
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: |
+          cd ./assets
+          yarn install
+          yarn deploy
+          yarn dump
+
+      - name: Cache local Elixir cache
+        uses: actions/cache@v2
+        with:
+          path: deps
+          key: ${{ runner.os }}-mix-${{ hashFiles('**/mix.lock') }}
+          restore-keys: ${{ runner.os }}-mix-
+
+      - name: Use Elixir
+        uses: actions/setup-elixir@v1
+        with:
+          elixir-version: "1.10.4"
+          otp-version: "23.0.2"
+
+      - name: Install and compile dependencies
+        run: mix do deps.get, deps.compile
+        env:
+          MIX_ENV: prod
+
+      - name: Digests and compresses static files
+        run: mix phx.digest
+        env:
+          MIX_ENV: prod
+
+      - name: Compile and release application
+        run: mix do compile, release
+        env:
+          MIX_ENV: prod
+      
+      - name: Create artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: backoffice
+          path: |
+            ./_build/prod/rel/backoffice
+            ./CHANGELOG.md
+            ./Dockerfile
+            ./rel/run.sh
+            ./version.txt
+  
+  test:
+    runs-on: ubuntu-latest
+    name: Test Elixir project
+
+    needs: [ build ]
```

Comment:
I don't think this is necessary. The only "shared" part is getting the Elixir dependencies. The rest ist distinct. I would propose to fetch the dependencies here as well and decouple the jobs.